ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -712,29 +753,38 @@ IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, long operationTimeMs)
    * @return the {@link IndexValue} of the ttl update record
```

Comment:
add comment for lifeVersion.
Just to confirm: -1 means the request comes from frontend, right?

Reply:
maybe  rename hasLifeVersion to isFrontendRequest

Reply:
right, -1 means it comes from the frontend, and if it has the lifeVersion defined it means it came from replication